src/app/features/user/forgot-password/forgot-password.component.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { async, ComponentFixture, TestBed, tick, fakeAsync, flush } from '@angular/core/testing';
+
+import { ForgotPasswordComponent } from './forgot-password.component';
+import { CustomMaterialModule } from '../../../core/material.module';
+import { ReactiveFormsModule, FormsModule } from '@angular/forms';
+import { SharedModule } from '../../../shared/shared.module';
+import { RouterTestingModule } from '@angular/router/testing';
+import { HomePageComponent } from '@app/features/home/home-page/home-page.component';
+import { HttpClientModule } from '@angular/common/http';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { AlertService } from '@app/core/services';
+
+import { UserService } from '../../../core/services/user/user.service';
+import { TestUserService } from '../../../core/services/testing/test-user.service';
+import { throwError } from 'rxjs';
+
+describe('ForgotPasswordComponent', () => {
+  let component: ForgotPasswordComponent;
+  let fixture: ComponentFixture<ForgotPasswordComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ ForgotPasswordComponent, HomePageComponent ],
+      imports: [
+        CustomMaterialModule,
+        ReactiveFormsModule,
+        FormsModule,
+        SharedModule,
+        HttpClientModule,
+        BrowserAnimationsModule,
+        RouterTestingModule.withRoutes([ { path: '', component: HomePageComponent } ])
+      ],
+      providers: [
+        {provide: UserService, useClass: TestUserService}
+      ]
+    })
+    .compileComponents().then(() => {
+      fixture = TestBed.createComponent(ForgotPasswordComponent);
+      component = fixture.componentInstance;
+      fixture.detectChanges();
+    });
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should set submmitted flag to true when submit form', () => {
+    component.onSubmit()
+    expect(component.submitted).toBeTruthy()
+  })
+
+  it('form should be invalid with email is empty', () => {
+    component.formControls.email.setValue('')
+    expect(component.forgotPassForm.invalid).toBeTruthy()
+  })
+
+  it('form should be invalid with length of invalid email format', () => {

```

Comment:
I think we should verify error message display correctly and what control is invalid because there are alot of reason form invalid